src/main/java/com/streamr/client/authentication/ChallengeAuthenticationMethod.java

```diff
@@ -0,0 +1,82 @@
+package com.streamr.client.authentication;
+
+import com.squareup.moshi.JsonAdapter;
+import com.streamr.client.StreamrClientOptions;
+import com.streamr.client.utils.HttpUtils;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.util.Date;
+
+import org.apache.commons.codec.DecoderException;
+import org.ethereum.crypto.ECKey;
+import org.apache.commons.codec.binary.Hex;
+import org.ethereum.crypto.HashUtil;
+import org.ethereum.util.ByteUtil;
+
+public class ChallengeAuthenticationMethod extends AuthenticationMethod {
```

Comment:
Since this is a user-facing class, let's consider its naming a bit more. 

The method involves an Ethereum key, a challenge, and a signature. To me as a user, the most important ingredient out of the three is the Ethereum key, and the others are implementation details.

So to me it feels that the class name should definitely contain "Ethereum". Possible choices for renaming:

- `EthereumAuthenticationMethod` (maybe this is descriptive enough?)
- `EthereumSignatureAuthenticationMethod`
- `EthereumChallengeAuthenticationMethod`